apps/project/serializers.py

```diff
@@ -459,10 +459,20 @@ class Meta:
         fields = '__all__'
         read_only_fields = ('project',)
 
+    def validate_project_usergroup(self, project, usergroup):
+        if ProjectUserGroupMembership.objects.filter(project=project, usergroup=usergroup).exists():
+            raise serializers.ValidationError({'usergroup': 'Usergroup already exist in the project'})
+
     def validate(self, data):
         data['project_id'] = int(self.context['view'].kwargs['project_id'])
+        usergroup = data.get('usergroup')
+        usergroup = self.validate_project_usergroup(data['project_id'], usergroup)
         return data
 
+    def create(self, validated_data):
+        instance, _ = ProjectUserGroupMembership.objects.get_or_create(**validated_data)
+        return instance
+
```

Comment:
This should be enough right?. Also for the patch method user group can be None.

```suggestion
    def validate(self, data):
        data['project_id'] = int(self.context['view'].kwargs['project_id'])
        usergroup = data.get('usergroup')
         if usergroup and ProjectUserGroupMembership.objects.filter(project=data['project_id'], usergroup=usergroup).exists():
            raise serializers.ValidationError({'usergroup': 'Usergroup already exist in the project'})
        return data
```

Reply:
Done!!